haskell/private/cc_libraries.bzl

```diff
@@ -241,13 +241,17 @@ def create_link_config(hs, posix, cc_libraries_info, libraries_to_link, binary,
 
     return (cache_file, static_libs, dynamic_libs)
 
+def _path_or_none(f):
+    if f != None:
+        return f.path
```

Comment:
Maybe this could go as a local definition inside in cc_library_key.

Reply:
Done.

Reply:
Never mind, this is not allowed in Bazel 4.0.0:
```
ERROR: /home/runner/.cache/bazel/_bazel_runner/3e6d0af2c61b2bb9852d7167e763fc51/external/rules_haskell/haskell/private/cc_libraries.bzl:247:5: nested functions are not allowed. Move the function to the top level.
ERROR: error loading package '': in /home/runner/.cache/bazel/_bazel_runner/3e6d0af2c61b2bb9852d7167e763fc51/external/rules_haskell/haskell/cabal.bzl: Extension 'haskell/private/cc_libraries.bzl' has errors
```
It seems to be allowed in Bazel HEAD though, the change worked when tested with `USE_BAZEL_VERSION="last_green"`.